### PR TITLE
psd estimation results page

### PR DESCRIPTION
**Pull request by ahnitz**
* move layout utilities to separate module and document them a bit better
* add a result page for the PSD estimation workflow.

 Untested at the moment, except to say that the workflow generator runs, and that the output folder / file layout looks right. I'll post back if/when I get a workflow to run.

Please take a look anyway, if you have time, in case something can be spotted wrong early.

### CODE REVIEW

File: pycbc/workflow/segment.py [hunk 1]
```diff
@@ -1129,6 +1129,7 @@ def get_analyzable_segments(workflow, out_dir, tags=[]):
         curr_file = OutSegFile(ifo, 'SCIENCE_OK', workflow.analysis_time,
                                curr_url, segment_list=sci_segs[ifo],
                                tags=tags)
+        curr_file.PFN(seg_ok_path, 'local')
```
Comment (spxiwh) [inline pycbc/workflow/segment.py]: I would really prefer it if you could move this into the toSegmentXml method in core.py

Comment (ahnitz) [inline pycbc/workflow/segment.py]: Ian,

The only reason I didn't was that this was done elsewhere many times, and
it is a failure to add the path twice, I believe. Without the time to check
and fix everywhere else, I wanted to make the minimal change. In principle,
I would actually like the toSegmentXml method to be replaced by the
segments_to_file function, which does this also. Add this to the list of
segment related cleanup.

Comment (spxiwh) [inline pycbc/workflow/segment.py]: Okay, as we're going to fix this properly post 1.2.0 anyway I won't argue on this point.

**Comment by ahnitz** [comment]
Ok, example output finished. 

https://atlas1.atlas.aei.uni-hannover.de/~ahnitz/LSC/ahnitz/O1/bns_tuning_er8a/pw2/html/

**Comment by duncan-brown** [comment]
@ahnitz any idea why the x-scale on the segment plots are messed up?

**Comment by ahnitz** [comment]
No idea.

**Comment by ahnitz** [comment]
To be a bit more verbose, however, I didn't touch that code, so I don't
know why.

**Comment by duncan-brown** [comment]
OK, I've seen that before. Must be a bug in the google code.

**Comment by duncan-brown** [comment]
This commit addresses part of https://github.com/ligo-cbc/pycbc/issues/388

Any chance you could patch to have the actual PSD result files copied in, so that they are available from the result page?

**Pull request merged**